tests/end2end-tests/graphbolt-gs-integration/graphbolt-training-inference.sh

```diff
@@ -0,0 +1,207 @@
+    -o | --output-path)
+        OUTPUT_PATH="${2-}"
+        shift
+        ;;
+    -?*) die "Unknown option: $1" ;;
+    *) break ;;
+    esac
+    shift
+    done
+
+    return 0
+}
+
+cleanup() {
+    trap - SIGINT SIGTERM ERR EXIT
+    # script cleanup here
+    if [[ -d "${OUTPUT_PATH}" ]]; then
+        echo "Cleaning up ${OUTPUT_PATH}"
+        rm -rf "${OUTPUT_PATH}"
+    fi
+}
+
+fdir_exists() {
+    # Take two args: first should be f or d, for file or directory
+    # second is the path to check
+
+    if [ "$1" == "f" ]
+    then
+        if [ ! -f "$2" ]
+        then
+            msg "$2 must exist"
+            exit 1
+        fi
+    elif [ "$1" == "d" ]
+    then
+        if [ ! -d "$2" ]
+        then
+            msg "$2 must exist"
+            exit 1
+        fi
+    else
+        msg "First arg to fdir_exists must be f or d"
+        exit 1
+    fi
+}
+
+parse_params "$@"
+
+GS_HOME=$(pwd)
+
+mkdir -p "$OUTPUT_PATH"
+cp -R "$INPUT_PATH" "$OUTPUT_PATH"
+
+# Ensure ip_list.txt exists and self-ssh works
+rm "$OUTPUT_PATH/ip_list.txt" &> /dev/null || true
+echo "127.0.0.1" > "$OUTPUT_PATH/ip_list.txt"
+ssh -o PreferredAuthentications=publickey -o StrictHostKeyChecking=no \
+    -p 2222 127.0.0.1 /bin/true || service ssh restart
+
+# Generate 1P LP data
+LP_INPUT_1P="${OUTPUT_PATH}/graphbolt-gconstruct-lp-1p"
+python3 -m graphstorm.gconstruct.construct_graph \
+    --add-reverse-edges \
+    --conf-file $GS_HOME/tests/end2end-tests/data_gen/movielens_lp.json \
+    --graph-name ml-lp \
+    --num-parts 1 \
+    --num-processes 1 \
+    --output-dir "$LP_INPUT_1P" \
+    --part-method random \
+    --use-graphbolt "true"
+
+LP_OUTPUT="$OUTPUT_PATH/gb-lp"
+msg "**************GraphBolt Link Prediction training. dataset: Movielens, RGCN layer 1, node feat: fixed HF BERT, BERT nodes: movie, inference: mini-batch, negative_sampler: joint, exclude_training_targets: false"
+python3 -m graphstorm.run.gs_link_prediction \
+    --cf $GS_HOME/training_scripts/gsgnn_lp/ml_lp.yaml \
+    --eval-frequency 300 \
+    --ip-config "$OUTPUT_PATH/ip_list.txt" \
+    --num-epochs 1 \
+    --num-samplers 0 \
+    --num-servers 1 \
+    --num-trainers 1 \
+    --part-config "$LP_INPUT_1P/ml-lp.json" \
+    --save-model-path "$LP_OUTPUT/model" \
+    --ssh-port 2222 \
+    --use-graphbolt true
+
+# Ensure model file was saved
+fdir_exists f "$LP_OUTPUT/model/epoch-0/model.bin"
+
+msg " **************GraphBolt Link Prediction embedding generation **************"
+
+python3 -m graphstorm.run.gs_gen_node_embedding \
+    --cf $GS_HOME/training_scripts/gsgnn_lp/ml_lp.yaml \
+    --eval-frequency 300 \
+    --inference \
+    --ip-config "$OUTPUT_PATH/ip_list.txt" \
+    --num-epochs 1 \
+    --num-samplers 0 \
+    --num-servers 1 \
+    --num-trainers 1 \
+    --part-config "$LP_INPUT_1P/ml-lp.json" \
+    --restore-model-path "$LP_OUTPUT/model/epoch-0" \
+    --save-embed-path "$LP_OUTPUT/embeddings" \
+    --ssh-port 2222 \
+    --use-graphbolt true
```

Comment:
Can we add another test with
``--exclude-training-targets True``, ``--reverse-edge-types-map user,rating,rev-rating,movie`` and GSgnnLPInBatchJointNegDataLoader and AllEtypeDistEdgeDataLoader? (These two dataloaders are customized dataloader.